cocotb/drivers/__init__.py

```diff
@@ -319,24 +323,24 @@ def _next_valids(self):
         on followed by the number of cycles to be off.
         The ``on`` cycles should be non-zero, we skip invalid generator entries.
         """
-        self.on = False
+        on = False
 
         if self.valid_generator is not None:
-            while not self.on:
```

Comment:
I feel like the point of these `on` an `off` properties is so the caller can inspect the current state of the object

Reply:
Indeed, e.g. the Avalon driver makes use of it.